odoo/modules/loading.py

```diff
@@ -130,10 +133,16 @@ def _load_data(cr, module_name, idref, mode, kind):
         model_names = registry.load(cr, package)
 
         loaded_modules.append(package.name)
-        if hasattr(package, 'init') or hasattr(package, 'update') or package.state in ('to install', 'to upgrade'):
+        if (hasattr(package, 'init') or hasattr(package, 'update')
+                or package.state in ('to install', 'to upgrade')):
+            models_updated |= set(model_names)
+            models_to_check -= set(model_names)
             registry.setup_models(cr)
             registry.init_models(cr, model_names, {'module': package.name})
             cr.commit()
+        elif package.state != 'to remove':
+            # no operation is performed on the module (install/upgrade/remove)
+            models_to_check |= set(model_names) & models_updated
```

Comment:
comment:
```python
# The current module has simply been loaded. The models extended by this module
# and for which we updated the schema, must have their schema checked again.
# This is because the extension may have changed the model (for instance, adding
# require=True on an existing field) but the schema has not been updated by this module.
```